src/app/beer_garden/api/http/schemas/v1/user.py

```diff
@@ -44,3 +44,12 @@ class UserPatchSchema(Schema):
 
     password = fields.Str()
     role_assignments = fields.List(fields.Nested(RoleAssignmentPatchSchema))
+
+
+class UserPasswordChangeSchema(Schema):
+    """Schema for changing a user's password"""
+
+    current_password = fields.Str(required=True)
+    new_password = fields.Str(
+        required=True, validate=validate.Length(min=1, error="Password required")
```

Comment:
The minimum length of 1 is just to ensure it's not possible to set the password to an empty string.  Any password rules more complicated than that requires a level of consideration that I wasn't interested in pursuing in the scope of this PR.